src/Servers/IIS/IIS/test/IIS.Tests/TlsHandshakeFeatureTests.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Authentication;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IIS.FunctionalTests;
+
+[SkipIfHostableWebCoreNotAvailable]
+[MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win8, SkipReason = "https://github.com/aspnet/IISIntegration/issues/866")]
+[SkipOnHelix("Unsupported queue", Queues = "Windows.Amd64.VS2022.Pre.Open;")]
+public class TlsHandshakeFeatureTests : StrictTestServerTests
+{
+    [ConditionalFact]
+    public async Task SetsTlsHandshakeFeature()
```

Comment:
ITlsHandshakeFeature should only be present for https. Add a negative test DoesNotSetTlsHandshakeFeatureForHttp.

Reply:
Added negative test